src/main/scala/uk/gov/nationalarchives/tdr/api/service/ClientFileMetadataService.scala

```diff
@@ -4,49 +4,59 @@ import java.sql.{SQLException, Timestamp}
+import uk.gov.nationalarchives.tdr.api.service.FileMetadataService._
+import uk.gov.nationalarchives.Tables.FilemetadataRow
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class ClientFileMetadataService(clientFileMetadataRepository: ClientFileMetadataRepository, uuidSource: UUIDSource)
+class ClientFileMetadataService(fileMetadataRepository: FileMetadataRepository,
+                                uuidSource: UUIDSource, timeSource: TimeSource)
                                (implicit val executionContext: ExecutionContext) {
 
-  def addClientFileMetadata(inputs: Seq[AddClientFileMetadataInput]): Future[Seq[ClientFileMetadata]] = {
-    val rows: Seq[ClientfilemetadataRow] = inputs.map(i => ClientfilemetadataRow(
-      uuidSource.uuid,
-      i.fileId,
-      i.originalPath,
-      i.checksum,
-      i.checksumType,
-      Timestamp.from(Instant.ofEpochMilli(i.lastModified)),
-      i.fileSize,
-      Timestamp.from(Instant.ofEpochMilli(i.datetime))))
-
-    clientFileMetadataRepository.addClientFileMetadata(rows).map(r => {
-      r.map(rowToOutput)
+  implicit class LongUtils(value: Long) {
+    def toTimestampString: String = Timestamp.from(Instant.ofEpochMilli(value)).toString
+  }
+
+  def addClientFileMetadata(inputs: Seq[AddClientFileMetadataInput], userId: UUID): Future[List[ClientFileMetadata]] = {
+    val time = Timestamp.from(timeSource.now)
+    val row: (UUID, String, Option[String]) => Tables.FilemetadataRow = FilemetadataRow(uuidSource.uuid, _, Option.empty, _, time, userId, _)
+    val inputRows = inputs.flatMap(input => {
+      List(
+        row(input.fileId, input.originalPath.getOrElse(""), Option(ClientSideOriginalFilepath)),
+        row(input.fileId, input.lastModified.toTimestampString, Option(ClientSideFileLastModifiedDate)),
+        row(input.fileId, input.fileSize.map(_.toString).getOrElse(""), Option(ClientSideFileSize)),
+        row(input.fileId, input.checksum.getOrElse(""), Option(SHA256ClientSideChecksum))
+      )
+    })
+    fileMetadataRepository.addFileMetadata(inputRows).map(rows => {
+      val fileToRow = rows.groupBy(f => f.fileid)
+      fileToRow.map {
+        case (fileId, rows) => convertToResponse(fileId, rows)
+      }.toList
     })
   }
 
   def getClientFileMetadata(fileId: UUID): Future[ClientFileMetadata] = {
-    clientFileMetadataRepository.getClientFileMetadata(fileId).map(_.head).map(rowToOutput).recover {
-      case nse: NoSuchElementException => throw InputDataException(s"Could not find metadata for file $fileId", Some(nse))
-      case e: SQLException => throw InputDataException(e.getLocalizedMessage, Some(e))
-    }
+    fileMetadataRepository.getFileMetadata(fileId, clientSideProperties: _*)
+      .map(rows => convertToResponse(fileId, rows))
+      .recover {
+        case nse: NoSuchElementException => throw InputDataException(s"Could not find metadata for file $fileId", Some(nse))
```

Comment:
Should the exception message be the following to make clear this is the client file metadata: "Could not find **client** metadata for file $FileId"

Reply:
Yeah, makes sense, I've changed it.